src/components/stack/index.tsx

```diff
@@ -1,26 +1,38 @@
 import React, { FC } from 'react';
-import { Stack as ChakraStack, StackProps } from '@chakra-ui/react';
+import {
+  Stack as ChakraStack,
+  StackProps as ChakraStackProps,
+} from '@chakra-ui/react';
 
-type Props = Pick<
-  StackProps,
+export { StackSeparator } from '@chakra-ui/react';
+
+export type StackProps = Pick<
+  ChakraStackProps,
   | 'align'
   | 'children'
   | 'direction'
   | 'justify'
-  | 'spacing'
+  | 'gap'
   | 'wrap'
-  | 'marginBottom'
-  | 'marginTop'
-  | 'divider'
+  | 'margin'
```

Comment:
Where does the requirement come from to extend all padding and margin props

Reply:
There's been a change to how `margins` and `paddings` behave. In v2 you could pass them as object containing sides:
```
margin={{ top: 'sm', bottom: 'lg'}
```
This doesn't work anymore; an unqualified one applies the same `margin` (or padding) on each side. So to keep the existing functionality I whitelisted all sides explicitly